lib/charliecloud.py

```diff
@@ -341,7 +368,17 @@ def layers_read(self):
          except tarfile.TarError as x:
             FATAL("cannot open: %s: %s" % (path, x))
          members = collections.OrderedDict([(m, None) for m in members_list])
-         layers[lh] = TT(fp, members)
+         if (lh in layers and members):
+            FATAL("duplicate non-empty layer %s" % lh[:7])
+         if (members):
```

Comment:
What is truth/falsity of a dictionary? Maybe be more explicit e.g. `if (len(members) > 0)`?